frontend/src/metabase-lib/expressions/process.js

```diff
@@ -31,13 +33,23 @@ export function processSource(options) {
     }
   };
 
-  const { source, startRule } = options;
+  const { source, query, stageIndex, startRule } = options;
 
-  let expression;
+  let expression = null;
+  let expressionClause = null;
```

Comment:
Adjusting to match `onChange` and `onCommit` props in `ExpressionEditorTextfield` which were/are typed with `null`.